src/func.d

```diff
@@ -1791,13 +1791,14 @@ public:
                         fbody = new CompoundStatement(Loc(), fbody, s);
                     }
                 }
-                if (returns && !fbody.isErrorStatement())
+                if (returns)
```

Comment:
Do we pay for the extra checks with slightly less performance in case this is done during speculative evaluations? Maybe just produce the complete @nogc coverage when not gagged.